### ISSUE TITLE
Instability during training

### ISSUE THREAD

**Issue by cerulean331** [datalogue/keras-attention#35]
I'm fairly new to this and for some reason I'm having ![crazy instabilities issues](http://139.82.47.36:8888/files/rrsayao/64%3B%20256%3B%2064%3B%2040%3B%2017%3B%20124%3B%200.2%3B%200.2%3B%20200%3B%20199%3B%20200%3B%200%3B%200.85423%3B%200.83730%3B%20titan%3B%200.8.jpg) during training. I've witnessed over 10% decrease in validation accuracy at some point. 

It's a many-to-many problem similar to pos tagging (vocab size much smaller). Input is an array of 40 integers (zero-padded), output is an array of 40 one-hot vectors. Any idea what I'm doing wrong?

```python
max_seqlen = 40
s_vocabsize = 17
t_vocabsize = 124

embed_size = 64
hidden_size = 128

input_ = Input(shape=(max_seqlen,), dtype='float32')
input_embed = Embedding(s_vocabsize, embed_size, input_length=max_seqlen , mask_zero=True)(input_)

bi_lstm = Bidirectional(LSTM(HIDDEN_SIZE, dropout=0.2, recurrent_dropout=0.2, return_sequences=True), merge_mode='concat')(input_embed)
dropout = Dropout(0.8)(bi_lstm)

y_hat = AttentionDecoder(hidden_size , alphabet_size=t_vocabsize, embedding_dim=embed_size )(dropout)

model = Model(inputs=input_, outputs=y_hat)
model.compile(loss='categorical_crossentropy', optimizer="adam", metrics=["accuracy"])```
